### ISSUE TITLE
Add require option for `require-jsdoc` to support constant exports for HoC

### ISSUE THREAD

**Issue by tarngerine** [gajus/eslint-plugin-jsdoc#1144]
## Motivation

Our codebase uses HoC (i.e. mobx observer) exports for our components, and we'd like `require-jsdoc` to properly flag a component export with no jsdoc comments:

<img width="753" alt="image" src="https://github.com/gajus/eslint-plugin-jsdoc/assets/1629901/2e4fd61c-1e63-4010-b087-18514ccb1ffa">

## Current behavior

It doesn't trigger for this HoC pattern with these options:

```
  "jsdoc/require-jsdoc": [
    "warn",
    {
      publicOnly: true,
      require: {
        ArrowFunctionExpression: true,
        FunctionDeclaration: true,
        FunctionExpression: true,
        ClassDeclaration: true,
        ClassExpression: true,
        MethodDefinition: true,
      },
      enableFixer: false,
    },
  ],
```

## Desired behavior

require-jsdoc should have an option for supporting regular assignment expressions with HoC

**Comment by brettz9** [comment]
You can add to the `contexts` option something like the following:

```js
          contexts: [
            'CallExpression[callee.name="observer"]',
          ],
```

To tweak the CallExpression further, you can introspect, using the `@typescript-eslint/parser`, on sample AST using https://astexplorer.net .

Closing as this should resolve, though feel free to comment further as needed

**Comment by tarngerine** [comment]
Perfect, thank you!
